src/encryption/aes-encryptor.ts

```diff
@@ -9,67 +9,160 @@ import { Logger } from "../logging/logger";
+        params.content = params.data.slice(16);
+
+        this.createHash(params);
+
+        return params;
+    }
 
-     Logger.log.debug("AesEncryptor.encrypt: start.");
+    /**
+     * Creates the key and IV using the passphrase and a random salt.
+     * @return {EncryptionParameters}        The encryption parameters.
+     */
+    public createKeyAndIV(): EncryptionParameters {
 
-     if (!this.key) {
-       throw new EncryptionError("Encryptionkey not available.", "AES256");
-     }
+        Logger.log.debug("AesIVDecryptor.createKeyAndIV: Start");
 
-     let cipher: Crypto.Cipher = Crypto.createCipher("aes256", this.key);
-     let encrypted: string = cipher.update(data, "utf8", "base64");
-     encrypted += cipher.final("base64");
+        let params: EncryptionParameters = new EncryptionParameters();
 
-     Logger.log.debug(`AesEncryptor.encrypt: encryption complete: '${encrypted}'.`);
+        // the salt is bytes 8-15.
+        params.salt = Crypto.randomBytes(8);
+        Logger.log.debug(`AesIVDecryptor.createKeyAndIV: Salt (hex): ${params.salt.toString("hex")}`);
 
-     return encrypted;
- }
+        this.createHash(params);
 
- /**
-  * Decrypts the object and then parses it as JSON.
-  */
- public decrypt(encrypted: string): string {
+        return params;
+    }
 
-     Logger.log.debug("AesEncryptor.decrypt: start.");
+    /**
+     * Creates the hash, given the salt and passphrase.
+     * @param {EncryptionParameters} params The encryption parameters.
+     */
+    private createHash(params: EncryptionParameters): void {
 
-     if (!this.key) {
-       throw new EncryptionError("Encryptionkey not available.", "AES256");
-     }
+        // the key and the IV are derived through 3 rounds of MD5 hashing, each generating 8 bytes of data.
+        let rounds: number = 3;
+        let data00: Buffer = Buffer.concat([new Buffer(this.passphrase, "utf8"), params.salt]);
+        let md5hash: Buffer[] = [new Buffer(Crypto.createHash("md5").update(data00).digest("base64"), "base64")];
 
-     try {
-       Logger.log.debug(`AesEncryptor.decrypt: encrypted data: ${encrypted}`);
+        for (let i: number = 1; i < rounds; i++) {
+            let data01: Buffer = Buffer.concat([md5hash[i - 1], data00]);
+            md5hash[i] = new Buffer(Crypto.createHash("md5").update(data01).digest("base64"), "base64");
+        }
 
-       let decipher: Crypto.Decipher = Crypto.createDecipher("aes256", this.key);
-       let decrypted: string = decipher.update(encrypted, "base64", "utf8");
-       decrypted += decipher.final("utf8");
+        params.key = Buffer.concat([md5hash[0], md5hash[1]]);
+        params.iv = md5hash[2];
 
-       Logger.log.debug(`AesEncryptor.decrypt: decrypted data: ${decrypted}`);
+        Logger.log.debug(`AesIVDecryptor.createHash: Key (hex): ${params.key.toString("hex")}`);
+        Logger.log.debug(`AesIVDecryptor.createHash: IV (hex): ${params.iv.toString("hex")}`);
```

Comment:
you sure you want this in the logger output? this will deffo end up all over ops logs.